### PR TITLE
feat: question discussion add/edit comments

### PR DESCRIPTION
**Pull request by thisislawatts**
Building on the work originally in #3100, this adds support for creating, editing and removing comments on Question content items.

PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

### CODE REVIEW

File: src/pages/Question/QuestionComments.tsx [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
-export const QuestionComments = ({ comments }: IProps) => {
+export const QuestionComments = ({
+  comments,
+  activeUser,
+  onSubmit,
+}: IProps) => {
+  const [comment, setComment] = useState('')
+
   return (
     <Box mt={5}>
```
Comment (mariojsnunes) [inline src/pages/Question/QuestionComments.tsx:22]: shouldn't this just use the `<DiscussionContainer />` component?

Comment (thisislawatts) [reply src/pages/Question/QuestionComments.tsx:22]: Thats right @mariojsnunes, I need to refactor based on the recent changes introduced by @benfurber :)

**Review by iSCJT — COMMENTED** [review]
Things I've noticed:

- Deleting a comment doesn't work for me, the database deletes but the UI is stuck on the confirmation dialog and requires a refresh.
- When adding a new comment if there are already multiple comments and the list is collapsed the new comment is not visible after submission, I think we should expand / scroll to the new comment.
- After adding a comment to your own question, refresh the page and the main question edit button seems to disappear.

File: src/pages/Question/QuestionComments.tsx [hunk 2]
```diff
@@ -1,16 +1,85 @@
...
+import { useDiscussionStore } from 'src/stores/Discussions/discussions.store'
+import { Card } from 'theme-ui'
 
-import type { IDiscussionComment } from 'src/models'
+import type { IDiscussion, IDiscussionComment } from 'src/models'
 
 interface IProps {
+  questionDocId: string
   comments: IDiscussionComment[]
+  activeUser: any
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/pages/Question/QuestionComments.tsx]: Resolved in d2019cae71b5fe0264411815e126abc24f466976

File: packages/components/src/CreateComment/CreateComment.test.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import { fireEvent } from '@testing-library/react'
+import { vi } from 'vitest'
+
+import { render } from '../tests/utils'
+import { CreateComment } from './CreateComment' // Adjust the import path as needed
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply packages/components/src/CreateComment/CreateComment.test.tsx]: Resolved in d2019cae71b5fe0264411815e126abc24f466976

**Comment by thisislawatts** [comment]
> After adding a comment to your own question, refresh the page and the main question edit button seems to disappear.
Going to pull this out to a separate issue.

**Review by iSCJT — APPROVED** [review]
LGTM

**Pull request merged**